discord/__init__.py

```diff
@@ -1,77 +1,117 @@
-
-__title__ = "disnake"
-__author__ = "Rapptz, EQUENOS"
-__license__ = "MIT"
-__copyright__ = "Copyright 2015-present Rapptz, 2021-present EQUENOS"
-__version__ = "2.3.0"
-
-__path__ = __import__("pkgutil").extend_path(__path__, __name__)
-
-import logging
-from typing import NamedTuple, Literal
-
-from .client import *
-from .appinfo import *
-from .app_commands import *
-from .user import *
-from .emoji import *
-from .partial_emoji import *
-from .activity import *
-from .channel import *
-from .guild import *
-from .flags import *
-from .member import *
-from .message import *
-from .asset import *
-from .errors import *
-from .permissions import *
-from .role import *
-from .file import *
-from .colour import *
-from .integrations import *
-from .invite import *
-from .template import *
-from .widget import *
-from .object import *
-from .reaction import *
-from . import utils, opus, abc, ui
-from .enums import *
-from .embeds import *
-from .mentions import *
-from .shard import *
-from .player import *
-from .webhook import *
-from .voice_client import *
-from .audit_logs import *
-from .raw_models import *
-from .team import *
-from .sticker import *
-from .stage_instance import *
-from .interactions import *
-from .components import *
-from .threads import *
-from .custom_warnings import *
-from .guild_scheduled_event import *
-
-
-class VersionInfo(NamedTuple):
-    major: int
-    minor: int
-    micro: int
-    releaselevel: Literal["alpha", "beta", "candidate", "final"]
-    serial: int
-
-
-version_info: VersionInfo = VersionInfo(major=2, minor=3, micro=0, releaselevel="beta", serial=0)
-
-logging.getLogger(__name__).addHandler(logging.NullHandler())
+import importlib
+import importlib.abc
+import importlib.machinery
+import importlib.util
+import os
+import sys
+import types
+
+MAIN_PACKAGE = "disnake"
+
+
+def _find_file(fullname, path=None):
+    """Give a fullname, find the file on disk."""
+    # this is hacky but it probably works?
+    # i couldn't figure out a better way to do this
+    # and i don't want to lie in my comments
+    # so i'm sorry
+    if path is None:
+        path = sys.path
+    for pth in path:
+        item = os.path.join(pth, fullname.replace(".", os.sep))
+        if os.path.exists(item):
+            return os.path.join(item, "__init__.py")
+        if os.path.exists(item + ".py"):
+            return item
+    raise ModuleNotFoundError("No module named '{}'".format(fullname))
+
+
+class MainLoader(importlib.abc.ExecutionLoader):
+    def get_filename(self, fullname):
+        spec = importlib.util.find_spec(fullname)
+        if not spec or not spec.origin:
+            raise ModuleNotFoundError("No module named '{}'".format(fullname))
+
+        return spec.origin
+
+    def get_source(self, fullname):
+        if file := self.get_filename(fullname):
+            with open(file, "r") as f:
+                return f.read()
+        raise ModuleNotFoundError("No module named '{}'".format(fullname))
```

Comment:
```suggestion
        with open(self.get_filename(fullname), "r") as f:
            return f.read()
```
`get_filename` can't return `None`

Reply:
https://github.com/DisnakeDev/discord-disnake/pull/3/commits/a7cbb406025e5a5ae96eb4eb12591da5c03103ff